chore: update ci pipeline and poetry version

argysamo opened the pull request:
- Introduce dependencies' caching, python black testing and pytest in ci pipeline. Set the ubuntu agent to an explicit version.
- Bump poetry version to 1.8.2, and bump all dev dependencies to their latest. Update flask to use <3.0.0 version. Following this version, the Markup module is not include in Flask anymore.
- Run `black -l 80` on project
- Update __version__ in __init__.py to "0.4.1" to make the test successful. (I don't see the functionality of this test. I will consider removing it in the next PR).

argysamo committed:
chore: update ci pipeline and poetry version

- Introduce dependencies' caching, python black testing and pytest in ci pipeline. Set the ubuntu agent to an explicit version.
- Bump poetry version to 1.8.2, and bump all dev dependencies to their latest. Update flask to use <3.0.0 version. Following this version, the Markup module is not include in Flask anymore.
- Run `black -l 80` on project
- Update __version__ in __init__.py to "0.4.1" to make the test successful. (I don't see the functionality of this test. I will consider removing it in the next PR).

marcuxyz commented on .github/workflows/ci.yaml:
```diff
@@ -6,26 +6,35 @@ jobs:
-          python -m pip install --upgrade pip
-          pip install -r requirements.txt
+      - name: Install dependencies
+        run: poetry install
```
You can put this command in makefile and run only 

E.g;

Make install

marcuxyz commented on .github/workflows/ci.yaml:
```diff
@@ -6,26 +6,35 @@ jobs:
         run: |
-          pip install black
-          black -l 80 --check
+          poetry run black -l 80 --check .
```
Here also...

argysamo committed:
chore: fix PR review comments

- Update Makefile to include all CI steps. Introduce `black` and `quality` tasks. The latter runs `black` and `pep8` tasks.
- Update ci.yaml to include make tasks include of raw commands
- Fix `black` related findings (i.e. mainly F401 ones). The F401 findings were related with typing module, so instructed `black` to ignore those.

argysamo commented:
Hey @marcuxyz ,
Thanks for your comments. I updated the make tasks to cover all CI steps and made some other minor updates. More details in commit message. 
What do you think?

marcuxyz commented on Makefile:
```diff
@@ -1,10 +1,15 @@
+.PHONY: test quality pep8 black types clean install build publish tree env
 
-test: pep8
-	py.test --cov=flask_googlemaps -l --tb=short --maxfail=1 tests/
```
was necessary to remove the `-l --tb=short --maxfail=1 tests/` flags?

marcuxyz commented:
> Hey @marcuxyz , Thanks for your comments. I updated the make tasks to cover all CI steps and made some other minor updates. More details in commit message. What do you think?

Very nice
Thank you @argysamo
I left **one comment** only.

marcuxyz commented:
@rochacbruno  please review the `pr`,
